test/membrane_rtmp_plugin/tcp_server_test.exs

```diff
@@ -0,0 +1,52 @@
+defmodule Membrane.RTMP.Source.TcpServer.Test do
+  use ExUnit.Case
+
+  alias Membrane.RTMP.Source.TcpServer
+
+  @port 9000
+  @local_ip "127.0.0.1" |> String.to_charlist() |> :inet.parse_address() |> elem(1)
+  @sample_data "Hello World"
+
+  test "TcpServer transfers the control to the process " do
+    server_options = [
+      port: @port,
+      tcp_options: [
+        :binary,
+        packet: :raw,
+        active: false,
+        reuseaddr: true,
+        ip: @local_ip
+      ],
+      serve_fn: fn socket ->
+        {:ok, receive_task} =
+          Task.start(fn ->
+            Process.whereis(__MODULE__) |> Process.link()
```

Comment:
Why not just save the pid as `caller = self()` before `server_options` instead of doing this whole registering? Also, why do you need to link the process?

Reply:
I registered the process probably for some debugging. I link the processes in order for the test to crash if the receiving process crashes.